src/Recruiter/Job/Repository.php

```diff
@@ -241,6 +241,181 @@ public function recentHistory($group = null, T\Moment $at = null, array $query =
         ];
     }
 
+    /**
+     * @param int
+     */
+    public function countSlowRecentJobs(
+        T\Moment $lowerLimit,
+        T\Moment $upperLimit,
+        $secondsToConsiderJobAsSlow=5
+    )
+    {
+        $archived = $this->archived->aggregate([
+            [
+                '$match' => [
+                    'last_execution.ended_at' => [
+                        '$gte' => T\MongoDate::from($lowerLimit),
+                    ],
+                ]
+            ],
+            [
+                '$project' => [
+                    '_id' => '$_id',
+                    'execution_time' => [
+                        '$subtract' => [
+                            '$last_execution.ended_at',
+                            '$last_execution.started_at'
+                        ]
+                    ],
+                ]
+            ],
+            [
+                '$match' => [
+                    'execution_time' => [
+                        '$gt' => $secondsToConsiderJobAsSlow*1000 
+                    ]
+                ],
+            ],
+        ])['result'];
+
+        return count($archived) + $this->scheduled->count([
+            'scheduled_at' => [
+                '$gte' => T\MongoDate::from($lowerLimit),
+                '$lte' => T\MongoDate::from($upperLimit)
+            ],
+            'last_execution.started_at' => [
+                '$exists' => true,
+            ],
+            'last_execution.ended_at' => [
+                '$exists' => true,
+            ],
+        ]);
```

Comment:
I understand the query on the `archived` collection, but I don't understand this query on the `scheduled` collection: why all failed scheduled jobs are considered slowly regardless their execution time?

Reply:
I agree with @silvadanilo the query should be almost the same for archived and scheduled

Reply:
Good point guys, I missed the `execution_time` check on the `scheduled` gonna fix it now